### PR TITLE
feat: use a horizontal ellipsis to truncate player name

### PR DESCRIPTION
**Pull request by LangLangBart**
### Description
- it might look better to cut off the player name if it is longer than the given length and add a horizontal ellipsis
 - since the rating is usually not seen anyway, i thought we could just remove it
 - see picture for more clarity

### CODE REVIEW

File: gui/common/functions_utility~autociv.js [hunk 1]
```diff
@@ -124,3 +124,21 @@ function brightenedColor(color)
...
+ */
+function truncateString(str, num)
+{
+    if (!autociv_NameSeenBefore[str])
+    {
+        let {nick} = splitRatingFromNick(str)
+        autociv_NameSeenBefore[str] = nick.length > num ? nick.slice(0, num-1) + "…" : nick;
+    }
+    return autociv_NameSeenBefore[str]
+}
```
Comment (nanihadesuka) [inline gui/common/functions_utility~autociv.js:144]: Move all this inside StatsOverlay class that is /session, no need to have it in /common

File: gui/common/functions_utility~autociv.js [hunk 2]
```diff
@@ -124,3 +124,21 @@ function brightenedColor(color)
...
+ * @param   {string}  str  [str description]
+ * @param   {number}  num  [num description]
+ * @return  {string}       Truncated string with a horizontal ellipsis
+ */
+function truncateString(str, num)
+{
+    if (!autociv_NameSeenBefore[str])
+    {
+        let {nick} = splitRatingFromNick(str)
+        autociv_NameSeenBefore[str] = nick.length > num ? nick.slice(0, num-1) + "…" : nick;
```
Comment (nanihadesuka) [inline gui/common/functions_utility~autociv.js:141]: Remove the tailing  "...",  just truncate. Is better to know more 3 letters of the name than having tree dots for this case.

**Review by nanihadesuka — CHANGES_REQUESTED** [review]
Changes needed

**Comment by LangLangBart** [comment]
I can do what you asked me to do. I just want to clarify that the "horizontal ellipsis" shall be removed.

It is only one character long, not three.

**Comment by LangLangBart** [comment]
- I wasn't sure if it made sense to just move the function to the class, since there wasn't much left after removing "Horizontal_ellipsis" from the patch.
- I thought I would try to maximize the usefulness by making the player name left-aligned, as this appear to look more common.

**Review by nanihadesuka — APPROVED** [review]

**Pull request merged**